includes/class-wp-security-bp-json.php

```diff
@@ -22,75 +21,54 @@
  * @subpackage wp_security_bp/includes
  * @author     Cyan Lovers <hello@cyanlove.com>
  */
-
 class WP_Security_BP_JSON {
 	/**
 	 * Final array to return formated for each case.
 	 *
 	 * @since    1.0.0
 	 * @access   protected
-	 * @var      array    $json    The array that will be returned to the admin (later transformed to JSON)
+	 * @var      array $json    The array that will be returned to the admin (later transformed to JSON)
 	 */
 	public $json = array();
 
-	/**
-	 * The unique identifier of this plugin.
-	 *
-	 * @since    1.0.0
-	 * @access   protected
-	 * @var      string    $plugin_name    The string used to uniquely identify this plugin.
-	 */
-	protected $plugin_name;
-
 	/**
 	 * Initialize the class and set its properties.
 	 *
 	 * @since    1.0.0
-	 * @param    string    $plugin_name       The name of this plugin.
 	 */
-	public function __construct( $plugin_name ) {
-
-		$this->plugin_name = $plugin_name;
+	public function __construct() {
 
 	}
 
 	/**
 	 * If the test is passed, returns formated information.
 	 *
 	 * @since    1.0.0
-	 * @param    string    $message       The message of OK.
+	 * @param    string $message       The message of OK.
+	 * @param    string $short_desc    Optional. The description of the check.
 	 */
-	public function pass( $short_desc = '', $message ) {
+	public function pass( $message, $args = [], $short_desc = '' ) {
 
 		$this->json['status']     = 'passed';
 		$this->json['short_desc'] = $short_desc;
 		$this->json['button']     = false;
-		$this->json['message']    = sprintf(
-			/* translators: %s: message, plugin name */
-			__(
-				'%s',
-				'%s'
-			),
-			$message,
-			$this->plugin_name
-		);
+		$this->json['message']    = $message;
 	}
 
-	public function fail( $short_desc = '', $message, $action = null ) {
+	/**
+	 * If the test is fail, returns formated information.
+	 *
+	 * @since    1.0.0
+	 * @param    string $message       The message of failure.
+	 * @param    string $short_desc    Optional. The description of the check.
+	 * @param    string $action        Optional. The action that fires the button.
+	 */
+	public function fail( $message, $args = [], $short_desc = '', $action = null ) {
 
 		$this->json['status']     = 'fail';
 		$this->json['short_desc'] = $short_desc;
 		$this->json['button']     = true;
 		$this->json['action']     = $action;
-		$this->json['message']    = sprintf(
-
-			/* translators: %s: message, plugin name */
-			__(
-				'%s',
-				'%s'
-			),
-			$message,
-			$this->plugin_name
-		);
+		$this->json['message']    = vsprintf( $message, $args );
```

Comment:
and here....
not good enough

Reply:
🤔 maybe this **does** the trick.

Reply:
it does... but lets be fair... it's a shit